soroban-env-host/src/host/metered_vector.rs

```diff
@@ -49,7 +49,7 @@ where
     A: MeteredClone,
 {
     // Constructs a empty new `MeteredVector`.
-    pub fn new() -> Self {
+    pub fn new() -> Result<Self, HostError> {
```

Comment:
Is the reason of changing this to a Result to be consistent with the other Vec construction functions? If so, maybe change `MeteredOrdMap::new()` to return Result as well?

Reply:
No, it's because it delegates to `Self::from_vec` which itself had to change here. Which is silly, I should just rewrite it to return `Self { vec: Vec::new() }`. I'll push a fix, thanks for pointing it out.

(Kinda wrote this one on autopilot)